pkg/logs/sds/scanner.go

```diff
@@ -23,6 +24,21 @@ const ScannedTag = "sds_agent:true"
 
 const SDSEnabled = true
 
+var (
+	tlmSDSConfiguredRules = telemetry.NewCounterWithOpts("sds", "rules_configured", []string{},
+		"Number of configured rules.", telemetry.Options{DefaultMetric: true})
```

Comment:
Did you consider using gauges for these rather than counters? It seems to me like users generally care about the _current_ number of configured rules, malformed rules, etc.

Reply:
There is multiple logs pipeline running in the Agent, a gauge would be a LWW of the last reconfigured pipeline. it's not ideal but these counts coupled with the reconfigurations counts could help getting an actual "live" rules, but, it's more about getting signals than about actively monitoring how many rules are enabled (signals as in, e.g., seeing the amount of errors growing).

Let me know if you think it's not making sense.

Reply:
Ah, yes. If there are multiple pipelines, I think we could also add a `pipeline` tag with an id. That could be helpful for differentiating.

Alternatively, if it is all-or-nothing (i.e. the counts from all pipelines should be the same) maybe that indicates that they should be reported at a higher level than the pipeline itself to avoid discrepancies.

Reply:
> Alternatively, if it is all-or-nothing (i.e. the counts from all pipelines should be the same)  maybe that indicates that they should be reported at a higher level than the pipeline itself to avoid discrepancies.

It was my first shot but it adds way more complexity than I'd like just to report this higher up (having to sync all concurrent calls reconfiguring the different pipelines). I can go with a pipeline ID which would be 0, 1, 2 , ... to avoid cardinality explosion. I'll give it a second shot and see what I can do to improve this here.